tests/test_handlers.py

```diff
@@ -0,0 +1,332 @@
+            ['[#7:1]#[#7:2]', 99., 99.],
+            ['[#7:1]-[#8X2:2]', 99., 99.],
+            ['[#7:1]~[#8X1:2]', 99., 99.],
+            ['[#8X2:1]-[#8X2:2]', 99., 99.],
+            ['[#16:1]-[#6:2]', 99., 99.],
+            ['[#16:1]-[#1:2]', 99., 99.],
+            ['[#16:1]-[#16:2]', 99., 99.],
+            ['[#16:1]-[#9:2]', 99., 99.],
+            ['[#16:1]-[#17:2]', 99., 99.],
+            ['[#16:1]-[#35:2]', 99., 99.],
+            ['[#16:1]-[#53:2]', 99., 99.],
+            ['[#16X2,#16X1-1,#16X3+1:1]-[#6X4:2]', 99., 99.],
+            ['[#16X2,#16X1-1,#16X3+1:1]-[#6X3:2]', 99., 99.],
+            ['[#16X2:1]-[#7:2]', 99., 99.],
+            ['[#16X2:1]-[#8X2:2]', 99., 99.],
+            ['[#16X2:1]=[#8X1,#7X2:2]', 99., 99.],
+            ['[#16X4,#16X3!+1:1]-[#6:2]', 99., 99.],
+            ['[#16X4,#16X3:1]~[#7:2]', 99., 99.],
+            ['[#16X4,#16X3:1]-[#8X2:2]', 99., 99.],
+            ['[#16X4,#16X3:1]~[#8X1:2]', 99., 99.],
+            ['[#15:1]-[#1:2]', 99., 99.],
+            ['[#15:1]~[#6:2]', 99., 99.],
+            ['[#15:1]-[#7:2]', 99., 99.],
+            ['[#15:1]=[#7:2]', 99., 99.],
+            ['[#15:1]~[#8X2:2]', 99., 99.],
+            ['[#15:1]~[#8X1:2]', 99., 99.],
+            ['[#16:1]-[#15:2]', 99., 99.],
+            ['[#15:1]=[#16X1:2]', 99., 99.],
+            ['[#6:1]-[#9:2]', 99., 99.],
+            ['[#6X4:1]-[#9:2]', 0.6, 0.7],
+            ['[#6:1]-[#17:2]', 99., 99.],
+            ['[#6X4:1]-[#17:2]', 99., 99.],
+            ['[#6:1]-[#35:2]', 99., 99.],
+            ['[#6X4:1]-[#35:2]', 99., 99.],
+            ['[#6:1]-[#53:2]', 99., 99.],
+            ['[#6X4:1]-[#53:2]', 99., 99.],
+            ['[#7:1]-[#9:2]', 99., 99.],
+            ['[#7:1]-[#17:2]', 99., 99.],
+            ['[#7:1]-[#35:2]', 99., 99.],
+            ['[#7:1]-[#53:2]', 99., 99.],
+            ['[#15:1]-[#9:2]', 99., 99.],
+            ['[#15:1]-[#17:2]', 99., 99.],
+            ['[#15:1]-[#35:2]', 99., 99.],
+            ['[#15:1]-[#53:2]', 99., 99.],
+            ['[#6X4:1]-[#1:2]', 99., 99.],
+            ['[#6X3:1]-[#1:2]', 99., 99.],
+            ['[#6X2:1]-[#1:2]', 99., 99.],
+            ['[#7:1]-[#1:2]', 99., 99.],
+            ['[#8:1]-[#1:2]', 99., 99.1]
+        ]
+
+
+        smirks = [x[0] for x in patterns]
+        params = np.array([[x[1], x[2]] for x in patterns])
+        hbh = bonded.HarmonicBondHandler(smirks, params)
+
+        mol = Chem.MolFromSmiles("C1CNCOC1F")
+
+        bond_idxs, (bond_params, bond_vjp_fn) = hbh.parameterize(mol)
+
+        assert bond_idxs.shape == (mol.GetNumBonds(), 2)
+        assert bond_params.shape == (mol.GetNumBonds(), 2)
+
+
+        bonded_param_adjoints = np.random.randn(*bond_params.shape)
+
+        # test that we can use the adjoints
+        ff_adjoints = bond_vjp_fn(bonded_param_adjoints)[0]
+
+        # if a parameter is > 99 then its adjoint should be zero (converse isn't necessarily true since)
+        mask = np.argwhere(bond_params > 90)
```

Comment:
90 or 99? Also below

Reply:
this is just me being sloppy, it's all the same.